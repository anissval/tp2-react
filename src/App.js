import {
    BrowserRouter, Link,
    Route, Switch
} from "react-router-dom";
import {Dashboard} from "./components/dashboard/Dashboard";
import {Heroes} from "./components/heroes/Heroes";
import {HeroDetails} from "./components/HeroDetails/HeroDetails";

export const App = (props) => {
    return (
        <BrowserRouter>

            <div>
                <h1>Tour of Heroes</h1>
                <ul className="header">
                    <li><Link to="/">Dashboard</Link></li>
                    <li><Link to="/heroes">Heroes</Link></li>
                </ul>
                <div className="content">
                    <Switch>
                        <Route exact path="/"><Dashboard/></Route>
                        <Route exact path="/dashboard"><Dashboard/></Route>
                        <Route path="/heroes"><Heroes/></Route>
                        <Route path="/heroDetails/:id"><HeroDetails/></Route>
                    </Switch>
                </div>
            </div>
        </BrowserRouter>
    );
}

