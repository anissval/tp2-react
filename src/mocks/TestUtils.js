// test-utils.js
import React from 'react'
import {render as rtlRender} from '@testing-library/react'
import {createStore} from 'redux'
import {Provider} from 'react-redux'

// my reducer
import {heroesListManager} from "../reducer";
import {BrowserRouter} from "react-router-dom";

const renderWithRouterAndRedux = (ui, {
    route = '/',
    initialState,
    store = createStore(heroesListManager, initialState),
    ...renderOptions
} = {}) => {
    window.history.pushState({}, 'Test page', route);

    const ReduxWrapper = ({children}) => {
        return <Provider store={store}><BrowserRouter>{children}</BrowserRouter></Provider>
    }
    return rtlRender(ui, {wrapper: ReduxWrapper, ...renderOptions});
};

const renderWithRedux = (ui, {
    initialState,
    store = createStore(heroesListManager, initialState),
    ...renderOptions
} = {}) => {
    const ReduxWrapper = ({children}) => {
        return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, {wrapper: ReduxWrapper, ...renderOptions});
};
// re-export everything
export * from '@testing-library/react'
// override render method
export {renderWithRouterAndRedux, renderWithRedux};
