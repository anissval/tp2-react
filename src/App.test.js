import React from 'react';
import {App} from "./App";
import {renderWithRedux} from "./mocks/TestUtils";
import {prettyDOM} from "@testing-library/react";

describe('App', () => {
  test('renders App component', () => {
    const tree = renderWithRedux(<App/>);
    console.log(prettyDOM(tree.container));
    expect(tree).toMatchSnapshot();
  });
});
