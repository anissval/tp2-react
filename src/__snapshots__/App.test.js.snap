// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          Tour of Heroes
        </h1>
        <ul
          class="header"
        >
          <li>
            <a
              href="/"
            >
              Dashboard
            </a>
          </li>
          <li>
            <a
              href="/heroes"
            >
              Heroes
            </a>
          </li>
        </ul>
        <div
          class="content"
        >
          <div>
            <h2>
              Top Heroes
            </h2>
            <div>
              <div>
                <a
                  href="/heroDetails/13"
                >
                  Bombasto
                </a>
              </div>
              <div>
                <a
                  href="/heroDetails/14"
                >
                  Celeritas
                </a>
              </div>
              <div>
                <a
                  href="/heroDetails/15"
                >
                  Magneta
                </a>
              </div>
              <div>
                <a
                  href="/heroDetails/16"
                >
                  RubberMan
                </a>
              </div>
              <div>
                <a
                  href="/heroDetails/17"
                >
                  Dynama
                </a>
              </div>
              <div>
                <a
                  href="/heroDetails/18"
                >
                  Dr IQ
                </a>
              </div>
              <div>
                <a
                  href="/heroDetails/19"
                >
                  Magma
                </a>
              </div>
              <div>
                <a
                  href="/heroDetails/20"
                >
                  Tornado
                </a>
              </div>
            </div>
            <label
              for="search"
            >
              Hero Search:
              <input
                id="search"
                name="search"
                placeholder="Hero name"
                type="text"
                value=""
              />
            </label>
            <ul>
              <li>
                <a
                  href="/hero-detail/13"
                >
                  Bombasto
                </a>
              </li>
              <li>
                <a
                  href="/hero-detail/14"
                >
                  Celeritas
                </a>
              </li>
              <li>
                <a
                  href="/hero-detail/15"
                >
                  Magneta
                </a>
              </li>
              <li>
                <a
                  href="/hero-detail/16"
                >
                  RubberMan
                </a>
              </li>
              <li>
                <a
                  href="/hero-detail/17"
                >
                  Dynama
                </a>
              </li>
              <li>
                <a
                  href="/hero-detail/18"
                >
                  Dr IQ
                </a>
              </li>
              <li>
                <a
                  href="/hero-detail/19"
                >
                  Magma
                </a>
              </li>
              <li>
                <a
                  href="/hero-detail/20"
                >
                  Tornado
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        Tour of Heroes
      </h1>
      <ul
        class="header"
      >
        <li>
          <a
            href="/"
          >
            Dashboard
          </a>
        </li>
        <li>
          <a
            href="/heroes"
          >
            Heroes
          </a>
        </li>
      </ul>
      <div
        class="content"
      >
        <div>
          <h2>
            Top Heroes
          </h2>
          <div>
            <div>
              <a
                href="/heroDetails/13"
              >
                Bombasto
              </a>
            </div>
            <div>
              <a
                href="/heroDetails/14"
              >
                Celeritas
              </a>
            </div>
            <div>
              <a
                href="/heroDetails/15"
              >
                Magneta
              </a>
            </div>
            <div>
              <a
                href="/heroDetails/16"
              >
                RubberMan
              </a>
            </div>
            <div>
              <a
                href="/heroDetails/17"
              >
                Dynama
              </a>
            </div>
            <div>
              <a
                href="/heroDetails/18"
              >
                Dr IQ
              </a>
            </div>
            <div>
              <a
                href="/heroDetails/19"
              >
                Magma
              </a>
            </div>
            <div>
              <a
                href="/heroDetails/20"
              >
                Tornado
              </a>
            </div>
          </div>
          <label
            for="search"
          >
            Hero Search:
            <input
              id="search"
              name="search"
              placeholder="Hero name"
              type="text"
              value=""
            />
          </label>
          <ul>
            <li>
              <a
                href="/hero-detail/13"
              >
                Bombasto
              </a>
            </li>
            <li>
              <a
                href="/hero-detail/14"
              >
                Celeritas
              </a>
            </li>
            <li>
              <a
                href="/hero-detail/15"
              >
                Magneta
              </a>
            </li>
            <li>
              <a
                href="/hero-detail/16"
              >
                RubberMan
              </a>
            </li>
            <li>
              <a
                href="/hero-detail/17"
              >
                Dynama
              </a>
            </li>
            <li>
              <a
                href="/hero-detail/18"
              >
                Dr IQ
              </a>
            </li>
            <li>
              <a
                href="/hero-detail/19"
              >
                Magma
              </a>
            </li>
            <li>
              <a
                href="/hero-detail/20"
              >
                Tornado
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
